File: 49. Group Anagrams.cpp
#include <unordered_map>
#include <algorithm>
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map <string, vector<string>> dic;
        for (int i = 0; i < strs.size(); i++)
        {
            string temp = strs[i];
            sort(temp.begin(), temp.end());
            if (dic.count(temp) == 0)
                dic[temp] = {};
            dic[temp].push_back(strs[i]);
        }
        vector <vector<string>> res;
        for(auto kv : dic)
            res.push_back(kv.second);
        return res;
    }
};

File: 49. Group Anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dic = {}
        for i in strs:
            temp = ''.join(sorted(i))
            if temp not in dic:
                dic[temp] = []
            dic[temp] += [i]
        res = []
        for key, value in dic.items():
            res += [value]
        return res
